File: csr_magic_frontend/src/constants/familyRelation.ts
import type { FamilyRelation } from '../types/participation';

export const FAMILY_RELATION_LABELS: Record<FamilyRelation, string> = {
  SPOUSE: '配偶',
  CHILD: '子女',
  PARENT: '父母',
  OTHER: '其他',
};

export const FAMILY_RELATION_OPTIONS: Array<{ value: FamilyRelation; label: string }> = [
  { value: 'SPOUSE', label: '配偶' },
  { value: 'CHILD', label: '子女' },
  { value: 'PARENT', label: '父母' },
  { value: 'OTHER', label: '其他' },
];

File: csr_magic_backend/src/main/java/com/csr/participation/entity/FamilyRelation.java
package com.csr.participation.entity;

/**
 * 家属与员工的亲属关系枚举
 * 用于活动报名时携带家属信息
 */
public enum FamilyRelation {
    /** 配偶 */
    SPOUSE,
    /** 子女 */
    CHILD,
    /** 父母 */
    PARENT,
    /** 其他亲属 */
    OTHER
}
